src/Products/Viewer/Cache/PngViewer.cs

```diff
@@ -0,0 +1,182 @@
+﻿using GroupDocs.Viewer.Options;
+using GroupDocs.Viewer.Results;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GroupDocs.Viewer.MVC.Products.Viewer.Cache
+{
+    class PngViewer : IDisposable
+    {
+        private readonly string filePath;
+        private readonly string htmlResourcePrefix;
+        private readonly IViewerCache cache;
+
+        internal readonly GroupDocs.Viewer.Viewer viewer;
+        private readonly PngViewOptions pngViewOptions;
+        private readonly ViewInfoOptions viewInfoOptions;
+        private static readonly Common.Config.GlobalConfiguration globalConfiguration = new Common.Config.GlobalConfiguration();
+
+        public PngViewer(string filePath, string htmlResourcePrefix, IViewerCache cache, int pageNumber = -1, int newAngle = 0)
+        {
+            this.cache = cache;
+            this.filePath = filePath;
+            this.htmlResourcePrefix = htmlResourcePrefix;
+            this.viewer = new GroupDocs.Viewer.Viewer(filePath);
+            this.pngViewOptions = this.CreatePngViewOptions(pageNumber, newAngle);
+            this.viewInfoOptions = ViewInfoOptions.FromPngViewOptions(this.pngViewOptions);
+        }
+
+        private PngViewOptions CreatePngViewOptions(int passedPageNumber = -1, int newAngle = 0)
+        {
+            PngViewOptions pngViewOptions = new PngViewOptions(pageNumber =>
```

Comment:
Codacy found an issue: [Rename 'pngViewOptions' which hides the field with the same name.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933705)